Added product id and sku in item adjustments.

ankurvr opened the pull request:
It can be useful to identify the option product when the commerce is integrated with ERP or CRM or any other system which requires SKU.

ankurvr committed:
Added product id and sku in item adjustments.

It can be useful to identify the option product when the commerce is integrated with ERP or CRM or any other system which requires SKU.

ankurvr commented:
@muzzwood May i know when this update will get merged and released for dowload?

Mark-H commented:
We'll give it a test and release it soon, likely next week.

ankurvr committed:
Added built child sku in item properties.

muzzwood commented on core/components/commerce_itemoptions/src/Modules/ItemOptions.php:
```diff
@@ -94,7 +97,7 @@ public function onAddToCart(Item $event): void
             $item->addPriceAdjustment($adjustment);
         }
-
+        $item->setProperty('item_childsku', join("-", $skuList));
```
This may be problematic because a product SKU can contain a hyphen.

ankurvr replied:
So what do you suggest as separator?

ankurvr replied:
Or we can exclude adding the same in item properties.

muzzwood replied:
Yeah, I'd prefer this not added to the `comOrderItem` properties. Perhaps I'm missing something; is there a particular reason you've added it? 

Referencing your first commit, you'd still be able to access the item option SKUs in the twig templates via the adjustments field. e.g.
```
{% for item in items %}
    {% for adjustment in item.adjustments %}
        {{ adjustment.properties.itemopt_product.sku }}
    {% endfor %}
{% endfor %}
```

BTW thanks for contributing! 🙂

ankurvr replied:
The reason: In e-commerce it's standard practice to generate parent product SKU using it's options SKU by gluing using hyphen, so i did. I also do Magento 2 so very well aware about it's practices w.r.t. SKUs.

In my current usecase it's not useful but i thought it may be helpful to community, also if it's already there then we won't have to run 2 loops to print such SKU.

So keeping it will also not harm anything i think.

Still if you want i can remove it right away.

muzzwood replied:
>The reason: In e-commerce it's standard practice to generate parent product SKU using it's options SKU by gluing using >hyphen, so i did. I also do Magento 2 so very well aware about it's practices w.r.t. SKUs.

Ah right, Commerce does the same when using a Product Matrix TV, but in that case it's generating the actual product SKU rather than adding a value to the properties. My worry is that developers may assume they can explode() this, which works until a user creates an incompatible SKU.

>In my current usecase it's not useful but i thought it may be helpful to community, also if it's already there then we won't >have to run 2 loops to print such SKU.

I guess it's similar to the argument concerning normalisation/denormalisation of db tables. In this case, it would typically be used inside the items loop anyway, so it's really just looping over a small adjustments array which would be negligible. 

If the second commit is reverted, then this has my vote for merging. Thanks a lot, @ankurvr!

ankurvr replied:
@muzzwood Now you can merge the PR. I just removed extra code related to parent product sku building.

ankurvr committed:
Removed auto-build parent product SKU feature. 

The auto build may become incompatible if user doesn't follow the best practice of creating SKUs for the options.

ankurvr commented:
@muzzwood Quick reminder if this is missed. :wink:

muzzwood commented:
Hi @ankurvr ,
Apologies for the delay, we've had our hands rather full preparing for MODXpo.
I've merged your PR, and it'll be included in the next release. Thanks again for your contribution! :)

Mark-H commented:
Sorry for the delay, this has now been released @ankurvr. Thanks for your contribution.
